src/output/plugins/PipeWireOutputPlugin.cxx

```diff
@@ -311,21 +311,29 @@ PipeWireOutput::PipeWireOutput(const ConfigBlock &block)
 void
 PipeWireOutput::SetVolume(float _volume)
 {
-	if (thread_loop == nullptr)
+	if (thread_loop == nullptr) {
+		// set volume before we are even enabled, restore it later
+		restore_volume = true;
+		volume = _volume;
 		return;
+	}
 
 	const PipeWire::ThreadLoopLock lock(thread_loop);
 
 	float newvol = _volume*_volume*_volume;
 
-	if (stream != nullptr && !restore_volume) {
+	if (stream != nullptr && !restore_volume && channels != 0) {
```

Comment:
What's the point of this new check? If `stream != nullptr`, then `channels` is also set.

Reply:
I observed a case where this clause would run with an uninitialized channel value leading to a stack corruption. I will attempt reproducing it again and get back to you

Reply:
Maybe the real problem is that `Enable()` doesn't initialize `stream`?

Reply:
I fixed that now in v0.23.x: 9975905fafcf49c829e7b79cd66e9da3aab509a8

Reply:
ah! good catch